rust/src/runner/context.rs

```diff
@@ -286,3 +275,243 @@ impl Drop for TestContext {
+                .into_iter()
+                .filter_map(|e| e.ok())
+                .collect();
+            assert_eq!(parent_content.len(), 1);
+
+            let content: Vec<_> = WalkDir::new(ctx.temp_dir.path())
+                .min_depth(1)
+                .max_depth(1)
+                .into_iter()
+                .filter_map(|e| e.ok())
+                .collect();
+            assert_eq!(content.len(), 1);
+            assert_eq!(content[0].path(), &file);
+
+            let file_stat = nix::sys::stat::lstat(&file).unwrap();
+            assert_eq!(
+                file_stat.st_mode & nix::libc::S_IFMT,
+                match ft {
+                    FileType::Dir => nix::libc::S_IFDIR,
+                    FileType::Fifo => nix::libc::S_IFIFO,
+                    FileType::Regular => nix::libc::S_IFREG,
+                    FileType::Socket => nix::libc::S_IFSOCK,
+                    FileType::Symlink(..) => nix::libc::S_IFLNK,
+                    _ => unimplemented!(),
+                }
+            )
+        }
+    }
+
+    #[test]
+    fn name_max() {
+        let tmpdir = TempDir::new().unwrap();
+        let settings = SettingsConfig { naptime: 0. };
+        let ctx = TestContext::new(&settings, &tmpdir.path());
+        let file = ctx.create_name_max(FileType::Regular).unwrap();
+        let name_len = file.file_name().unwrap().to_string_lossy().len();
+
+        let max_len = pathconf(ctx.base_path(), nix::unistd::PathconfVar::NAME_MAX)
+            .unwrap()
+            .unwrap();
+        assert_eq!(name_len, max_len as usize);
+        let mut invalid = file.clone();
+        invalid.set_file_name(invalid.file_name().unwrap().to_string_lossy().into_owned() + "x");
+
+        assert_eq!(
+            chmod(&invalid, Mode::empty()).unwrap_err(),
+            Errno::ENAMETOOLONG
+        );
+    }
+
+    #[test]
+    fn path_max() {
+        let tmpdir = TempDir::new().unwrap();
+        let settings = SettingsConfig { naptime: 0. };
+        let ctx = TestContext::new(&settings, &tmpdir.path());
+        let file = ctx.create_path_max(FileType::Regular).unwrap();
+        let path_len = file.to_string_lossy().len();
+
+        // including null char
+        let max_len = pathconf(ctx.base_path(), nix::unistd::PathconfVar::PATH_MAX)
+            .unwrap()
+            .unwrap()
+            - 1;
+        assert_eq!(path_len, max_len as usize);
+        let mut invalid = file.clone();
+        invalid.set_file_name(invalid.file_name().unwrap().to_string_lossy().into_owned() + "x");
+
+        assert_eq!(
+            chmod(&invalid, Mode::empty()).unwrap_err(),
+            Errno::ENAMETOOLONG
+        );
+    }
+
+    #[test]
+    fn new_file() {
+        let current_umask = nix::sys::stat::umask(Mode::from_bits_truncate(ALLPERMS));
+        nix::sys::stat::umask(current_umask);
+
+        for ft in [
+            FileType::Regular,
+            FileType::Dir,
+            // FileType::Fifo,
```

Comment:
Did you mean to leave these comments here?

Reply:
Yes, I want to write tests for those file types, but they require specific handling.